lacework/resource_lacework_alert_profile.go

```diff
@@ -0,0 +1,219 @@
+	response, err := lacework.V2.Alert.Profiles.Create(alertProfile)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("fields", setAlertProfileFields(response.Data.Fields))
+	log.Printf("[INFO] Created alert profile with id %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkAlertProfileRead(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+		response api.AlertProfileResponse
+	)
+
+	log.Printf("[INFO] Reading alert profile with id %s\n", d.Id())
+	err := lacework.V2.Alert.Profiles.Get(d.Id(), &response)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("id", response.Data.Guid)
+	d.Set("extends", response.Data.Extends)
+	d.Set("alerts", response.Data.Alerts)
+	d.Set("fields", setAlertProfileFields(response.Data.Fields))
+
+	log.Printf("[INFO] Read alert profile with id %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkAlertProfileUpdate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+		alerts   []api.AlertTemplate
+	)
+	profileID := d.Get("name").(string)
+
+	err := castSchemaSetToArrayOfAlertTemplate(d, "alert", &alerts)
+	if err != nil {
+		return err
+	}
+	log.Printf("[INFO] Updating alert profile %s with data:\n%+v\n", profileID, alerts)
+
+	response, err := lacework.V2.Alert.Profiles.Update(profileID, alerts)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("fields", setAlertProfileFields(response.Data.Fields))
+	log.Printf("[INFO] Updated alert profile with id %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkAlertProfileDelete(d *schema.ResourceData, meta interface{}) error {
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Deleting alert profile with id %s\n", d.Id())
+	err := lacework.V2.Alert.Profiles.Delete(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Deleted alert profile with id %s\n", d.Id())
+	return nil
+}
+
+func importLaceworkAlertProfile(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	var response api.AlertProfileResponse
+	lacework := meta.(*api.Client)
+
+	log.Printf("[INFO] Importing Lacework Alert Profile with id: %s\n", d.Id())
+
+	if err := lacework.V2.Alert.Profiles.Get(d.Id(), &response); err != nil {
+		return nil, fmt.Errorf(
+			"unable to import Lacework resource. Alert Profile with id '%s' was not found",
+			d.Id(),
+		)
+	}
+	log.Printf("[INFO] Alert Profile found with id: %s\n", response.Data.Guid)
+	return []*schema.ResourceData{d}, nil
+}
+
+func castSchemaSetToArrayOfAlertTemplate(d *schema.ResourceData, attr string, templateList *[]api.AlertTemplate) (err error) {
+	var (
+		t    api.AlertTemplate
+		list []any
+	)
+
+	if d.Get(attr) == nil {
+		return errors.Errorf("attribute %s not found", attr)
+	}
+
+	list = d.Get(attr).(*schema.Set).List()
+	for _, item := range list {
+		val := sanitizeAlertTemplateKeys(item.(map[string]interface{}))
```

Comment:
there is nothing that enforces that `item` IS actually a `map[string]interface{}` to this code could actually panic. I recommend doing this instead:
```suggestion
		iMap, ok := item.(map[string]interface{})
		if !ok {
			// maybe even print a log or something?
			continue
		}
		val := sanitizeAlertTemplateKeys(iMap)
```

Reply:
👍